src/nix/main.cc

```diff
@@ -92,6 +92,19 @@ static bool haveInternet()
 #endif
 }
 
+static void disableNet()
+{
+    // FIXME: should check for command line overrides only.
+    if (!settings.useSubstitutes.overridden)
+        settings.useSubstitutes = false;
```

Comment:
@roberth so your idea would be to have `useSubstitute`, `tarballTtl`, `fileTransfersettings` etc check in their getters if we are in offline mode?